### ISSUE TITLE
Survival Mode Balance Changes

### ISSUE THREAD

**Issue by Delocuro** [MinestrapTeam/Minestrappolation-4#115]
While Werty's working on updating to Forge 1.9, I'll be doing some survival gameplay testing.I'll be making a list here of balance changes to make after the 1.9 update, so if anyone has suggestions or comments from their own survival experiences, feel free to post them here.
- [ ] Upon first entering a world (not after every re-spawn), players should spawn with a full set of "Ragged Armor". This specialty armor is not craftable, gives slightly better protection than Leather Armor, but has very low durability, causing it to break very quickly. The intent is that this will allow players to survive long enough to get some materials built up, as currently 1.9 skeletons can 2-shot any player if they use the default starting health and are playing on hard mode, but will not provide the player with any sort of permanent protection since the armor breaks so fast.
- [ ] Radiant Quartz should be slightly more common to account for its extra uses.
- [ ] Alloy Furnaces should only consume twice as much fuel as normal Furnaces, whereas they currently consume four times as much.
- [ ] Wing Sinew needs some additional uses, or needs to be less commonly dropped, as it tends to pile up quickly after you've gotten one Hang Glider.
- [ ] Withered Bones and Slime Cores should both be slightly rarer,
- [ ] Meurodite should be slightly rarer.
- [ ] Torite should be slightly more common.
- [ ] Sorting Pipes should give three per craft instead of two.
- [ ] Ghast Tears should be found slightly more commonly while sifting Soul Sand.
- [ ] Quantum Claimerators should prevent mobs from destroying the terrain within their range (i.e.; Creeper explosions don't destroy blocks, Endermen can't pick things up, Silverfish can't crawl into blocks, etc).
- [ ] Quantum Claimerators should be slightly cheaper.
- [ ] Dimensional Fisheries' chance to drop Netherrack from a block of Compressed Netherrack should be far lower -- at the moment, you can effectively infinitely farm nether items after mining only about a stack of Netherrack.
- [ ] Compressed Blocks should break slightly faster when placed on the Dimensional Fishery. Not really a balance change, dimensional fishing just feels a bit tedious at the moment.

**Comment by medsal15** [comment]
Maybe add a fertilizer made from organic materials with a composter. Less powerful than bone meal, but easier to get.

**Comment by Delocuro** [comment]
@medsal15 
Hmm... That could actually get a bit overpowered, seeing as we do already have like four different ways of Bone-Mealing things (you can craft Bonemeal from both Bones and Animal Bones, and you can also bone-meal plants using Guano or a Torite Hoe).

What might make more sense balance-wise is making the fertilizer an item that you would apply to the soil itself, rather than the plants. That way you could have things like soils that auto-bonemeal their crops over time, or soils that make the plants growing on them drop more crops when harvested, etc.

Anyways, I'll go ahead and add that to a new issue for new features to add after updating to 1.9, as it's more of a feature thing than a balance change.

**Comment by djdor11** [comment]
Here's what I suggest:
1. As I suggested once before, make Quantum Claimerators useful in singleplayer - make them basically protect an area not only from other players like multiplayer, but also from mobs.
   For example in their area, creepers won't cause destruction, zombies won't be able to break doors, and endermen won't be able to steal blocks.
   That would be especially useful in protecting villages or maintaining areas without the need for light and fencing. But you'd probably want to increase their range to make it more effective.
   I especially recommend this for the sake of consistency with item/blocks usefulness. Having this one block that is only useful in multiplayer yet is craftable in singleplayer doesn't make much sense to me.
2. Your own villager changes? Villager related things is one of the only things you didn't give attention to with this mod, I'm sure you have your own ideas for them but the one thing I can think of is having a way of make them follow you. Considering there is a way to make pretty much every other passive mob in the game follow you (with food etc), I don't see why you couldn't make villagers follow you when you hold emeralds. Emeralds only exist because villagers exist and it would only make sense that villagers will actively follow you if you hold them, considering that's what they literally pursue in the game. Villagers are almost always needed in bases, especially if the only village you find is incredibly far, and the only way to get them to places now is hell quite literally.
3. I don't recommend making the game easier like you suggested, Minecraft is already too easy. Keep in mind most people who seek mods like Minestrapp are people who are fed up with vanilla and would like some challenge and new things to work for and experience in the game.
   If anything I would make the game harder. Minestrapp's massive addition of new items and gadgets already makes the game easier so think about it. The only 'easy' thing I would recommend adding is more tooltips for items. I noticed that some items in the mod and vanilla minecraft has tooltips that explain them and some don't. I believe every item in the game should have a tooltip, both for consistency and for eliminating the need for constant wiki searching. If you're interested, I'm willing to write the missing tooltips for all the items in vanilla and Minestrapp. For example:
   Dirt:
   Building Material.
   Used for farming.
   Seeds:
   Farming Material.
   Used for growing wheat.
   Chickens seek them.

I really hope you consider those suggestions, I am an old school fan of Minecraft and Minestrapp is one of the few mods that I think expand the game yet keep it vanilla enough while improving it.

**Comment by Delocuro** [comment]
@djdor11 

Definitely like the Claimerator idea, so I'll add that to the list! Dunno how soon that'll be done though -- preventing mob griefing in a localized area could get a little tricky to code, but we'll take a stab at it and see if we can get it to work. Increasing their range could cause some issues though -- currently, each claimerator claims exactly one chunk, so making it larger would cause issues like figuring out what happens when multiple claimerator claims overlap eachother, etc. What we'll probably end up doing is just making them slightly less expensive, since they're already a bit more difficult to obtain in survival multiplayer than we'd like.

As for villagers, I've wanted to add our own villager trades for a while, but there's some Forge limitations that have been hindering it; specifically that fact that we can only add trades by coding in our own custom villagers, rather than adding trades to existing ones. That method would work for some trades, but for others we'd have to do weird stuff that clashes with Vanilla, like adding duplicate farmer villagers just to allow them to buy and sell Minestrapp crops. Supposedly there's some more villager hooks in Forge for 1.9 though, so we'll try looking into it again once we update to that.

And yeah, I see your point on the base heart increase, but the main reason for that is to avoid unnecessary early-game tedium -- a skeleton in 1.9 on normal difficulty can knock you down to 1 heart in 2 hits if you start at 5, so it's nearly impossible to survive for more than a day unless you just bury yourself in the ground every night and sit there doing absolutely nothing until all the mobs have died the next morning. What I might consider instead is having the player start off with some sort of starter-armor that gives decent protection but very low durability -- that way they'd at least have a chance of survival before getting some form of shelter built, but the protection increase wouldn't last the whole game like the starting health increase would.

Finally, I really like the tooltip idea, but I'm not sure if we can add extra tooltips and things to vanilla items without overwriting vanilla files, so we'll have to check on that first. One thing I've wanted to add for a while to sort of fill that wiki void is something called the Minechiridion. It'd basically be a book that starts out with no pages and can be accessed via a button on the player's inventory screen. The player could then find a new item called a Lost Page by either sifting Rubble or looting dungeon chests, which, when right clicked, would fill in a randomly selected page of the Minechiridion. Each page would contain all the information about one block or item -- how it is crafted, where it is naturally found, what it can be used for, what tools can be used to mine it, etc -- so by exploring the world and finding more pages the player would slowly unlock information about all of the blocks and items in the game and have access to those at any time.

So far we haven't had time to start coding it yet, but if you have any suggestions on the mechanics before we start, we'd love to hear them -- it's an idea I haven't really seen in any other mods before, so feedback would definitely be useful. :)

**Comment by djdor11** [comment]
Regarding the starting armor thing - if that's what you got in mind you might as well just add leather armor to the vanilla bonus chest, that way the whole difficulty thing would be optional and no need for new game mechanics and new types of armor.

Regarding tooltips - I think it should be possible to add tooltips to vanilla items in forge, I saw other mods do such things (think Hunger Overhaul).

The Minechiridion idea is awesome, but instead of making the pages completely random I suggest making them as very rare drops from mobs and blocks, and of course some could be found in dungeon chests, as long as where they are found is related to what information they give you.
A different way to add a 'wiki' to the game is instead of the books and pages thing you can have a button in your inventory which shows a gui with every item you got so far (think the stats menu), with bars that fill up as you acquire more of that item, with stages separations in the bar. Let's say the first stage is 16, (from the beginning of that world you acquired 16 of that item) you get information about minor stuff you can craft with that item, then say stage 2 is 32, more major crafting recipes, and then 64 the bar is full and you receive all the recipes of that item and maybe other information.

That's about it, I'm glad you're taking interest in my ideas and I'll be glad to keep coming up with them and of course seeing them implemented. I'm looking forward to the mod items villager trades as this would be fantastic but again consider a game mechanic which allows villagers to follow you.

**Comment by Delocuro** [comment]
Yeah, the bonus chest idea would probably be a good middleground on difficulty. I will probably plan on still adding the new armor set though -- leather armor lasts a pretty decently long time early in the game, so it could get overpowered adding a full set to the bonus chest. I'll also have to take a look at how hunger overhaul does their tooltips -- if they overwrite the vanilla food items to add their hunger mechanics that could be an issue, but otherwise it'd be a good place to start from in terms of code.

I like the idea of having the pages be found in locations that relate to their respective blocks and items, but that could get difficult to do for things that are only crafted such as Furnaces, Bricks, Fences, etc. Going off of your "page-leveling" idea though, one option could be to just unlock each page as soon as you've obtained one of its respective item. The initial unlocked page would simply show an image of the block/item and its name, and further information would unlock as you obtain more of that item, such as recipes, uses, locations for obtaining it, block properties such as luminosity and blast-resistance, etc. The quantity of each block/item needed per level of info would have to vary based on the item, though, since otherwise things like Dirt would fill in all their information right at the start of the game while things like Beacons would be nearly impossible to completely fill in.

And as for the Villager following thing, I like the idea, but unfortunately that one really isn't feasible. We'd have to overwrite the Villagers' basic AI class file, which would require a more complicated process for installing the mod and would make it incompatible with pretty much any mod that adds or alters villagers in any way. If we can find a way of doing it without overwriting vanilla files we'll try it, but I can't really guarantee when/if that would happen...

**Comment by djdor11** [comment]
I absolutely agree with your perspective on the page-leveling idea. I can see it looking like this - you get an item, you see a message saying "Page get! (Name)"  and once you open this page, you see as you said the image and name plus basic information (especially if tooltips won't be possible) and then greyed out buttons of "Uses", "Recipes" and "Technical Information". When you hover over the buttons you see an overlay saying "Unlocked at - [GOAL]".
But there should also be pages for mobs of course (when you kill/tame them you get the "Page get") which will have to be a bit different than the blocks and items but the same basic system.

Anyway regarding the villagers I think I saw a forge mod doing that, it definitely won't hurt trying. Though I would say Minstrappolation holds up enough on its own so keeping complete compatibility with other mods isn't very necessary in my opinion. If you're that eager on compatibility though, you could have 2 releases of Minestrappolation for each version - "Lite" and "Full". Lite being 100% compatible with other mods but limited features, full being 100% feature packed but limited compatibility.
